lib/internal/test_runner/mock/mock.js

```diff
@@ -32,17 +34,19 @@ function kDefaultFunction() {}
 
 class MockFunctionContext {
   #calls;
-  #mocks;
   #implementation;
   #restore;
   #times;
+  #queue;
+  #emptyImpl;
 
   constructor(implementation, restore, times) {
     this.#calls = [];
-    this.#mocks = new SafeMap();
     this.#implementation = implementation;
     this.#restore = restore;
     this.#times = times;
+    this.#queue = [];
```

Comment:
I haven't reviewed this new version, but it looks complex. The problem with your previous implementation was that it used the size of the`#mocks` map without considering the mocks that were scheduled in the future. I wonder if you could have looped over the map to determine that [here](https://github.com/nodejs/node/pull/49088/commits/4bb76565d9bda0ad56dc532bf0be8d10eb1a944d#diff-87fe4fbfd23996c49072c2fb88829aaa0ed1be188b9da9d072471b0a523cd1c8R68). If that would work, it would be a much simpler change.

Reply:
you are right, fixed